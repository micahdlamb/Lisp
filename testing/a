()
NIL
(defun N () ())
(n)
(defun N () NIL)
(n)
1 2 3
(quote (7 . nil))
(quote (NIL . NIL))
(quote (1 2 3 4 . 5))
(quote (() 2 3 4 . 5))
(quote (1 2 3 4 . NIL))
(quote (() 2 3 4 . NIL))

(plus -5 +5)

(defun xx NIL 5)
(defun yy () 5)
(xx)
(yy)
(xx 1)

(defun x (a b c d)
	(cond
		((greater a b) (PLUS (car (quote (5 9))) (car (cdr (quote (234 632))))))
		((x d c b (PLUS a 1)) (plus d (plus b (REMAINDER a c))))
	)
)

(defun inner (a b)
	(times (plus (plus a e) f) b)
)

(defun outer (e f)
	(cond
		((NULL (inner 1 2)) (inner 52 325))
		((EQ (inner 432 123) (inner 1 2)) (inner 41 32))
		((atom (inner 1 2)) (inner 532 421))
	)
)

(cons
	(times (quotient (outer 63 32) (x 352 432 441 231)) (minus (x 32 344 3 34) (outer 234 32)))
	NIL
)


(defun yyy (a b c d)
	(cons
		(cons
			(cons
				(cons
					(car (cdr (cons (quote t) (cons 5 (INT (NULL (eq a T)))))))
					(quote (NULL NIL))
				)
				(cons
					(less c d) (quote a)
				)
			)
			NIL
		)
		T
	)
)

(yyy 1 2 3 4)

(x 5 10 15 20)
(defun gay (x) (x x 10 15 20))
(gay 5)
car
(car 5)
(cdr 5)
(defun 5 (1 2 3) 5)
(5 1 2 3)
(MINUS 10 5)
(PLUS 10 -5)

(eq 5 5)
(atom 5)
(atom NIL)
(int 12)
(eq NIL NIL)
(eq (less (plus 5 10) 5) (less 2 1))

(atom (quote (x . y)))
(atom (greater 10 5))
(eq (less (plus 5 10) 5) (less 5 10))
(eq NIL T)
(eq 5 NIL)
(eq 5 10)
(eq NIL 5)
(int NIL)
(int T)
(int (less 10 5))

(defun T (x y z) x)
(x)
(cond NIL)
(defun z (a) (a 10 20))
(z outer)
(x 1 2 3 4 5)
(x 1 2 3)
(eq x x)
(yyy (eq T NIL) (Atom (greater 5 1)) (NULL T) (int (plus 5 6)))
(add T 5)
(plus T 5)
(times 5 NIL)
(divide x 5)
(quotient y 10)
(remainder (atom NULL))
(z 10)
((car (quote (times plus))) 5 5)
(cond (1 2 3))
(plus . x)
(defun tits (a -b)
(5 .5)
(5. 5)
(defun * (a b) x)
(defun -y (x) x)
(defun +x (x) x)
(x .)
(. x)
)
(
.
.