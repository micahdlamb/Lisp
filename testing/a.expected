NIL
NIL
N
NIL
N
NIL
1
2
3
(7)
(NIL)
(1 . (2 . (3 . (4 . 5))))
(NIL . (2 . (3 . (4 . 5))))
(1 2 3 4)
(NIL 2 3 4)
0
XX
YY
5
5
ERROR: XX called with 1 args instead of 0
X
INNER
OUTER
(-87250280)
YYY
((((5 NULL NIL) . (T . A))) . T)
35
GAY
35
ERROR: undefined variable: CAR
ERROR: CAR called with arg of invalid type
ERROR: CDR called with arg of invalid type
ERROR: DEFUN called with arg of invalid type
ERROR: first item of S-EXPR must be function: 5 found
5
5
T
T
T
T
T
T
NIL
NIL
NIL
NIL
NIL
NIL
NIL
NIL
NIL
NIL
ERROR: DEFUN called with arg of invalid type
ERROR: X called with 0 args instead of 4
ERROR: COND arg 0 must be a list of size 2
Z
ERROR: undefined variable: OUTER
ERROR: X called with 5 args instead of 4
ERROR: X called with 3 args instead of 4
ERROR: undefined variable: X
ERROR: LESS called with args of invalid type
ERROR: undefined function: ADD
ERROR: PLUS called with args of invalid type
ERROR: TIMES called with args of invalid type
ERROR: undefined function: DIVIDE
ERROR: undefined variable: Y
ERROR: REMAINDER called with 1 args instead of 2
ERROR: undefined function: A
ERROR: first item of S-EXPR must be function: (CAR (QUOTE (TIMES PLUS))) found
ERROR: COND arg 0 must be a list of size 2
ERROR: S-EXPR beginning with ID is function call and must be a list
ERROR: - must be followed by a number
